### ISSUE TITLE
Narrowing conversion compiler error with latest VS 2019 (cl 19.23.28107)

### ISSUE THREAD

**Issue by patrikhuber** [ceres-solver/ceres-solver#525]
Hi,

Building with the latest VS 2019 (v16.3.10 - this is the latest official release, **not** Preview), `cl.exe` `19.23.28107 for x64`, results in a compiler error in `ceres-solver\internal\ceres\expression_ref.cc` for me:

```
Severity	Code	Description	Project	File	Line	Suppression State	Suppression State
Error	C2398	Element '1': conversion from 'ceres::internal::ExpressionId' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=ceres::internal::ExpressionRef
        ]	ceres	ceres-solver\internal\ceres\expression_ref.cc	128		

```

That's on this line:

```
ExpressionRef Ternary(ComparisonExpressionRef c,
                      ExpressionRef a,
                      ExpressionRef b) {
  return MakeFunctionCall("ternary", {c.id, a.id, b.id}); // <--
}
```

**Comment by patrikhuber** [comment]
I can fix it and successfully compile Ceres by explicitly "casting":

```
return MakeFunctionCall("ternary", {ExpressionRef(c.id), ExpressionRef(a.id), ExpressionRef(b.id)});
```

But I have no idea whether that makes sense or perhaps creates broken code.

**Comment by patrikhuber** [comment]
@darglein I think this might be related to the new autodiff codegen.

Would it be possible to add CI testing on the latest stable MSVC for Ceres? I think that would be a huge help for many people.

**Comment by darglein** [comment]
Thank you for posting this problem.
It should be fixed with ea057678c5f603289380e0c2975c477693653c29.

**Comment by sandwichmaker** [comment]
@ale
